### PR TITLE
[Keyboard] Add Borderland

### PR DESCRIPTION
**Pull request by kiwikey**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/kiwikey/borderland/config.h [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+/*== or choose animations ==*/
+//#    define RGBLIGHT_EFFECT_BREATHING
+//#    define RGBLIGHT_EFFECT_RAINBOW_MOOD
+//#    define RGBLIGHT_EFFECT_RAINBOW_SWIRL
+//#    define RGBLIGHT_EFFECT_SNAKE
+//#    define RGBLIGHT_EFFECT_KNIGHT
+//#    define RGBLIGHT_EFFECT_CHRISTMAS
+//#    define RGBLIGHT_EFFECT_STATIC_GRADIENT
+//#    define RGBLIGHT_EFFECT_RGB_TEST
+//#    define RGBLIGHT_EFFECT_ALTERNATING
```
Comment (drashna) [inline keyboards/kiwikey/borderland/config.h:60]: Could you explicitly define which animations are enabled? 
```suggestion
#define RGBLIGHT_EFFECT_BREATHING
#define RGBLIGHT_EFFECT_RAINBOW_MOOD
#define RGBLIGHT_EFFECT_RAINBOW_SWIRL
#define RGBLIGHT_EFFECT_SNAKE
#define RGBLIGHT_EFFECT_KNIGHT
#define RGBLIGHT_EFFECT_CHRISTMAS
#define RGBLIGHT_EFFECT_STATIC_GRADIENT
#define RGBLIGHT_EFFECT_RGB_TEST
#define RGBLIGHT_EFFECT_ALTERNATING
#define RGBLIGHT_EFFECT_TWINKLE
```

File: keyboards/kiwikey/borderland/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,22 @@
+// Copyright 2021 KiwiKey (@KiwiKey)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include QMK_KEYBOARD_H
+#include "encoder.c"
```
Comment (drashna) [inline keyboards/kiwikey/borderland/keymaps/default/keymap.c:5]: C files shouldn't be included this way. 
```suggestion
```

File: keyboards/kiwikey/borderland/keymaps/default/encoder.c [hunk 3]
```diff
@@ -0,0 +1,13 @@
+// Copyright 2021 KiwiKey (@KiwiKey)
```
Comment (drashna) [inline keyboards/kiwikey/borderland/keymaps/default/encoder.c:1]: This file should be removed.

File: keyboards/kiwikey/borderland/keymaps/via/encoder.c [hunk 4]
```diff
@@ -0,0 +1,13 @@
+// Copyright 2021 KiwiKey (@KiwiKey)
```
Comment (drashna) [inline keyboards/kiwikey/borderland/keymaps/via/encoder.c:1]: This file should be removed.

File: keyboards/kiwikey/borderland/keymaps/via/keymap.c [hunk 5]
```diff
@@ -0,0 +1,38 @@
+// Copyright 2021 KiwiKey (@KiwiKey)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include QMK_KEYBOARD_H
+#include "encoder.c"
```
Comment (drashna) [inline keyboards/kiwikey/borderland/keymaps/via/keymap.c:5]:
```suggestion
```

File: keyboards/kiwikey/borderland/borderland.c [hunk 6]
```diff
@@ -0,0 +1,4 @@
+// Copyright 2021 KiwiKey (@KiwiKey)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "borderland.h"
```
Comment (drashna) [inline keyboards/kiwikey/borderland/borderland.c:4]: This allows for a default behavior to be defined, without the need for the user to specify the behavior.  This is very useful for the configurator which doesn't have a way to configure this behavior, currently.

```suggestion
#include "borderland.h"

#ifdef ENCODER_ENABLE
bool encoder_update_kb(uint8_t index, bool clockwise) {
    if (!encoder_update_user(index, clockwise)) { return false; }
    if (index == 0) {
        if (clockwise) {
            tap_code_delay(KC_VOLU, 10);
        } else {
            tap_code_delay(KC_VOLD, 10);
        }
    }
    return true;
}
#endif
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by kiwikey** [comment]
Hi @drashna, thank you for reviewing the codes. I have updated and commited changes as your suggestion.
I also re-built firmwares in my local PC, and tested with PCB, they all works.

Please help to re-check again and merge my PR if there is no issues left.

P.s: I am not so familiar with Git, hope all gonna be fine. Please let me know if there is need of any correction.

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Comment by kiwikey** [comment]
Thank you guys!!! Happy weekend.

**Pull request merged**